src/logic/logic.ts

```diff
@@ -149,8 +160,48 @@ function readPushByteOp(
   return [size, byteArray];
 }
 
+/** sanityCheckProgram performs heuristic program validation:
+ * check if passed in bytes are Algorand address or is B64 encoded, rather than Teal bytes
+ *
+ * @param program - Program bytes to check
+ * @throws
+ * @returns
+ */
+export function sanityCheckProgram(program: Uint8Array) {
+  if (!program || program.length === 0) throw new Error('empty program');
+
+  const lineBreakOrd = '\n'.charCodeAt(0);
+  const blankSpaceOrd = ' '.charCodeAt(0);
+  const tildeOrd = '~'.charCodeAt(0);
+
+  const isPrintable = (x: number) => blankSpaceOrd <= x && x <= tildeOrd;
+  const isAsciiPrintable = program.every(
+    (x: number) => x === lineBreakOrd || isPrintable(x)
+  );
+
+  if (isAsciiPrintable) {
+    const programStr = Buffer.from(program.buffer).toString();
```

Comment:
You should use `Buffer.from(program)` instead of `Buffer.from(program.buffer)`. It's possible that the `Uint8Array` `program` represents only a substring of the underlying `.buffer` object.

Reply:
this actually blows my mind, ok I will just pass in `program` then.